src/Agent.Worker/TaskManager.cs

```diff
@@ -155,26 +155,22 @@ private async Task DownloadAsync(IExecutionContext executionContext, Pipelines.T
             var configurationStore = HostContext.GetService<IConfigurationStore>();
             AgentSettings settings = configurationStore.GetSettings();
             Boolean signingEnabled = (settings.SignatureVerification != null && settings.SignatureVerification.Mode != SignatureVerificationMode.None);
+            Boolean alwaysExtractTask = signingEnabled || settings.AlwaysExtractTask;
 
-            if (File.Exists(destDirectory + ".completed") && !signingEnabled)
+            if (File.Exists(destDirectory + ".completed") && !alwaysExtractTask)
```

Comment:
Do we set this to true automatically when signing is enabled?

Reply:
Yes, check out line 158.